contracts/eth/nearprover/utils/eth-ledger-signer.js

```diff
@@ -0,0 +1,111 @@
+const hid = require('@ledgerhq/hw-transport-node-hid').default;
+const defaultPath = "m/44'/60'/0'/0/0";
+function waiter(duration) {
+    return new Promise((resolve) => {
+        setTimeout(resolve, duration);
+    });
+}
+class EthLedgerSigner extends ethers.Signer {
+    constructor(provider, path) {
+        super();
+        if (!path) {
+            path = defaultPath;
+            console.log(defaultPath)
+        }
+        ethers.utils.defineReadOnly(this, 'path', path);
+        ethers.utils.defineReadOnly(this, 'provider', provider || null);
+        ethers.utils.defineReadOnly(this, '_eth', hid.create().then((transport) => {
+            const eth = new hwAppEth(transport);
+            return eth.getAppConfiguration().then((_config) => {
+                return eth;
+            }, (error) => {
+                return Promise.reject(error);
+            });
+        }, (error) => {
+            return Promise.reject(error);
+        }));
+    }
+    _retry(callback, timeout) {
+        return new Promise(async (resolve, reject) => {
+            if (timeout && timeout > 0) {
+                setTimeout(() => {
+                    reject(new Error('timeout'));
+                }, timeout);
+            }
+            const eth = await this._eth;
+            // Wait up to 5 seconds
+            for (let i = 0; i < 50; i++) {
+                try {
+                    const result = await callback(eth);
+                    return resolve(result);
+                }
+                catch (error) {
+                    if (error.id !== 'TransportLocked') {
+                        return reject(error);
+                    }
+                }
+                await waiter(100);
+            }
+            return reject(new Error('timeout'));
+        });
+    }
+    async getAddress() {
+        const account = await this._retry((eth) => eth.getAddress(this.path));
+        return ethers.utils.getAddress(account.address);
+    }
+    async signMessage(message) {
+        if (typeof message === 'string') {
+            message = ethers.utils.toUtf8Bytes(message);
+        }
+        const messageHex = ethers.utils.hexlify(message).substring(2);
+        const sig = await this._retry((eth) => eth.signPersonalMessage(this.path, messageHex));
+        sig.r = '0x' + sig.r;
+        sig.s = '0x' + sig.s;
+        return ethers.utils.joinSignature(sig);
+    }
+    async signTransaction(transaction) {
+        const tx = await ethers.utils.resolveProperties(transaction);
+        const baseTx = {
+            chainId: tx.chainId || undefined,
+            data: tx.data || undefined,
+            gasLimit: tx.gasLimit || undefined,
+            gasPrice: tx.gasPrice || undefined,
+            nonce: tx.nonce ? ethers.BigNumber.from(tx.nonce).toNumber() : undefined,
+            maxFeePerGas: tx.maxFeePerGas || undefined,
+            maxPriorityFeePerGas: tx.maxPriorityFeePerGas || undefined,
```

Comment:
Please add the comment which will explain that these lines has been added as they haven't been yet accepted in the original repo, providing an appropriate link to PR

Reply:
fixed at e5f44756a4da2b1b886ed1e94421532ed5fa1d42